src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -247,20 +247,27 @@ protected Object readResolve() {
     	return this;
     }
     
-    private Map<String, String> getScanProperties(Run<?,?> build, TaskListener listener) {
-    	VariableResolver<String> resolver = build instanceof AbstractBuild ? new BuildVariableResolver((AbstractBuild<?,?>)build, listener) : null;
-    	Map<String, String> properties = m_scanner.getProperties(resolver);
-		properties.put(CoreConstants.SCANNER_TYPE, m_scanner.getType());
-        properties.put(CoreConstants.APP_ID,  m_application);
-        properties.put(CoreConstants.SCAN_NAME, resolver == null ? m_name : Util.replaceMacro(m_name, resolver) + "_" + SystemUtil.getTimeStamp()); //$NON-NLS-1$
-		properties.put(CoreConstants.EMAIL_NOTIFICATION, Boolean.toString(m_emailNotification));
-                properties.put("FullyAutomatic", Boolean.toString(!m_intervention));
-		properties.put("APPSCAN_IRGEN_CLIENT", "Jenkins");
-		properties.put("APPSCAN_CLIENT_VERSION", Jenkins.VERSION);
-		properties.put("IRGEN_CLIENT_PLUGIN_VERSION", JenkinsUtil.getPluginVersion());
-		properties.put("ClientType", JenkinsUtil.getClientType());
-		return properties;
-    }
+  
+    private Map<String, String> getScanProperties(Run<?,?> build, TaskListener listener) throws AbortException {
+
+		VariableResolver<String> resolver = build instanceof AbstractBuild ? new BuildVariableResolver((AbstractBuild<?,?>)build, listener) : null;
+		try {
+			Map<String, String> properties = m_scanner.getProperties(resolver);
+			properties.put(CoreConstants.SCANNER_TYPE, m_scanner.getType());
+			properties.put(CoreConstants.APP_ID, m_application);
+			properties.put(CoreConstants.SCAN_NAME, resolver == null ? m_name : Util.replaceMacro(m_name, resolver) + "_" + SystemUtil.getTimeStamp()); //$NON-NLS-1$
+			properties.put(CoreConstants.EMAIL_NOTIFICATION, Boolean.toString(m_emailNotification));
+      properties.put("FullyAutomatic", Boolean.toString(!m_intervention));
+			properties.put("APPSCAN_IRGEN_CLIENT", "Jenkins");
+			properties.put("APPSCAN_CLIENT_VERSION", Jenkins.VERSION);
+			properties.put("IRGEN_CLIENT_PLUGIN_VERSION", JenkinsUtil.getPluginVersion());
+			properties.put("ClientType", "jenkins-" + SystemUtil.getOS() + "-" + JenkinsUtil.getPluginVersion());
+			return properties;
+		} 
+		catch(NullPointerException e) {
```

Comment:
Do we know what object could be null and lead to a null pointer exception? It's generally better to check for null than to catch nullpointerexceptions.
Also, for the message in the catch block, can you use the Messages class so that the message can be translated?

Reply:
"m_scanner" was getting null while doing the upgrade testing means running the pre-configured mobile analyzer job in this new scenario Jenkins.
Okay, i will use the message class for the message that is being returned and will let you know.

Reply:
Thanks @vishalhcl-5960. Instead of catching the NullPointerException, can you instead do a check to see if m_scanner == null and in that case throw the AbortException? That would be a better approach over catching a NullPointerException, since the latter could cause us to miss bugs during testing (i.e. we catch some other NullPointerException, incorrectly allowing a test to pass).